bootcamp_project/Module 8/More_OOP_Concepts/VideoGame_Store_OOP_EMira1.ts

```diff
@@ -0,0 +1,78 @@
+class VideoGame {
+
+    // Static attributes
+    static availableGames: VideoGame[] = [];
+    static soldOutGames: VideoGame[] = [];
+    static totalSold: number;
+
+
+    // Instance attributes
+    title: string;
+    developer: string;
+    availableCopies: number;
+    price: number;
+    copiesSold: number;
+
+    constructor(title: string, developer: string, availableCopies: number, price: number, copiesSold: number, availableGames: string[]) {
+        this.title = title;
+        this.developer = developer;
+        this.availableCopies = availableCopies;
+        this.price = price;
+        this.copiesSold = copiesSold;
+
+        VideoGame.availableGames.push(this);
+    }
+
+    // Instance methods
+    isAvailable(): boolean {
+    return this.availableCopies > 0
+    }
+
+    sell(numberOfCopies:number = 1): number {
+
+        try {
+            if ( this.isAvailable() ) {
+                this.availableCopies -= numberOfCopies;
+                this.copiesSold += numberOfCopies;
+                VideoGame.totalSold = VideoGame.totalSold + this.copiesSold;
+
+                if ( this.availableCopies === 0 ) {
+                    VideoGame.soldOutGames.push(this);
+                    VideoGame.availableGames = VideoGame.availableGames.filter(game => game != this);
+                }
+            }
+
+            return this.price;
```

Comment:
I think this price should be multiplied by the number of copies being sold:
```suggestion

            return this.price * numberOfCopies;
```